src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -88,103 +138,1165 @@ static DamageTable D_80A31B1C = {
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A304B8.s")
+        Actor_SetFocus(thisx, 15.0f);
+        if (this->effectAlpha > 0.0f) {
+            if (this->damageEffectType != ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) {
+                Math_StepToF(&this->effectAlpha, 0.0f, 0.05f);
+                this->effectScale = (this->effectAlpha + 1.0f) * 0.2f;
+                this->effectScale = CLAMP_MAX(this->effectScale, 0.4f);
+            } else if (Math_StepToF(&this->frozenSteamScale, 0.4f, 0.01f) == 0) {
+                func_800B9010(thisx, 0x20B2);
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A3072C.s")
+void EnSlime_Draw(Actor* thisx, PlayState* play) {
+    s32 i;
+    EnSlime* this = THIS;
+    Vec3f wobbleScale;
+    Color_RGBA8* primColor;
+    Color_RGBA8* envColor;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A30778.s")
+    OPEN_DISPS(play->state.gfxCtx);
+    func_8012C28C(play->state.gfxCtx);
+    func_8012C2DC(play->state.gfxCtx);
+    func_800B8118(&this->actor, play, 0);
+    if (this->iceBlockTimer != 0xFF) {
+        gSPSegment(POLY_XLU_DISP++, 10, D_801AEFA0);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0xAA, 0xFF, 0xFF, 0xFF, 0xFF);
+        gDPSetEnvColor(POLY_XLU_DISP++, 0x96, 0xFF, 0xFF, this->iceBlockTimer);
+    } else {
+        primColor = &sPrimColors[this->actor.params];
+        envColor = &sEnvColors[this->actor.params];
+        AnimatedMat_Draw(play, sSlimeAnimTex);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0x64, primColor->r, primColor->g, primColor->b, primColor->a);
+        gDPSetEnvColor(POLY_XLU_DISP++, envColor->r, envColor->g, envColor->b, 0xFF);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A30820.s")
+    if (this->actionFunc == EnSlime_UpdateDamageReaction) {
+        wobbleScale.x = 1.0f - (sin_rad((f32)this->timer * 1.5707964f) * 0.3f);
+        wobbleScale.y = (sin_rad((f32)this->timer * 1.5707964f) * 0.3f) + 1.0f;
+        wobbleScale.z = 1.0f - (cos_rad((f32)this->timer * 1.5707964f) * 0.3f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A30924.s")
+        Matrix_RotateXFApply(this->wobbleRot.x);
+        Matrix_RotateZF(this->wobbleRot.z, MTXMODE_APPLY);
+        Matrix_Scale(wobbleScale.x, wobbleScale.y, wobbleScale.z, MTXMODE_APPLY);
+        Matrix_RotateZF(-this->wobbleRot.z, MTXMODE_APPLY);
+        Matrix_RotateXFApply(-this->wobbleRot.x);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A30944.s")
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_XLU_DISP++, gChuchuBodyDL);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A309C8.s")
+    if (this->iceBlockTimer == 0xFF) {
+        Scene_SetRenderModeXlu(play, 0, 1U);
```

Comment:
```suggestion
        Scene_SetRenderModeXlu(play, 0, 1);
```
You usually don't need the trailing U except for rare matching cases (applies a few places)